### PR TITLE
Added Color Selectors

### PR DESCRIPTION
**Pull request by Ryan10145**
Adds color selector for #134

#### Changes proposed in this pull request:
- Adds in two color selectors in the settings page, one for the top bar and one for the background
- Allows the following colors, with more colors easily added: default (uses light/dark theme), red, blue, black, brown, green

#### Screenshots for the change: 

gif

### CODE REVIEW

**Comment by sk9331657** [comment]
@Akshat-Jain  This can be closed as I have already claimed this task and this task is for gci .

File: src/scripts/options.js [hunk 1]
```diff
@@ -311,12 +323,12 @@ function saveOptions(e) {
 
 			var selectedtheme = document.querySelector("#theme").value;
 
-			var url = BASE_URL+"/aaa/changeUserSettings.json?key1=theme&value1="+selectedtheme+"&access_token="+accessToken+"&count=1";
+			var themeUrl = BASE_URL+"/aaa/changeUserSettings.json?key1=theme&value1="+selectedtheme+"&access_token="+accessToken+"&count=1";
 
 			// fire the api call to change settings value on server
 			$.ajax({
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 2]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
 
+	}
+	else if(nameOfSettingsChanged === "top-bar-color"){
+		browser.storage.sync.set({
+			topBarColor: document.querySelector("#top-bar-color").value
+		});
+
+		//check if user is logged in
+		if (accessToken != "") {
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 3]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+			var selectedTopBarColor = document.querySelector("#top-bar-color").value;
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 4]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+			var topBarUrl = BASE_URL + "/aaa/changeUserSettings.json?key1=topBarColor&value1=" + selectedTopBarColor + 
+				"&access_token=" + accessToken + "&count=1";
+
+			// fire the api call to change settings value on server
+			$.ajax({
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 5]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+				url: topBarUrl,
+				dataType: "jsonp",
+				jsonpCallback: "q",
+				jsonp: "callback",
+				crossDomain: true,
+				success: function (response) {
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 6]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+					alert("Top bar color updated successfuly.");
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 7]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+				}
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 8]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+		}
 	}
+	else if (nameOfSettingsChanged === "message-pane-color") {
+		browser.storage.sync.set({
+			messagePaneColor: document.querySelector("#message-pane-color").value
+		});
+
+		//check if user is logged in
+		if (accessToken != "") {
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 9]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+			var selectedMessagePaneColor = document.querySelector("#message-pane-color").value;
 
+			var messagePaneUrl = BASE_URL + "/aaa/changeUserSettings.json?key1=messagePaneColor&value1=" + selectedMessagePaneColor +
+				"&access_token=" + accessToken + "&count=1";
+
+			// fire the api call to change settings value on server
+			$.ajax({
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 10]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+				url: messagePaneUrl,
+				dataType: "jsonp",
+				jsonpCallback: "q",
+				jsonp: "callback",
+				crossDomain: true,
+				success: function (response) {
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 11]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+					alert("Message pane color updated successfuly.");
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 12]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+				}
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

File: src/scripts/options.js [hunk 13]
```diff
@@ -331,9 +343,69 @@ function saveOptions(e) {
...
+			});
+
```
Comment (realslimshanky) [inline src/scripts/options.js]: Remove this unnecessary line.

**Comment by Ryan10145** [comment]
@realslimshanky Just to clarify, when you say to remove unnecessary lines, are you talking about the empty lines or something else?

**Comment by realslimshanky** [comment]
@Ryan10145 just the empty lines.

**Comment by Ryan10145** [comment]
Ok, I've removed the empty lines

**Comment by Ryan10145** [comment]
@akshatnitd I've done a bit of testing, and I found that the dark theme text is very hard to see when there is a custom background color. Other than that, I don't see any other issues that jump out to me. I've changed a few things to improve this contrast, and I've included a gif to demonstrate how this version performs on my computer. 

gif3

**Comment by Ryan10145** [comment]
@akshatnitd Ok, I misunderstood how the color selector should work. I looked at the chrome extension and just copied how that worked. I'll change it to how you say it should work.

**Comment by Ryan10145** [comment]
I've changed the color selector accordingly, here's new footage of it:

gif4

Let me know if there are further issues.

**Pull request merged**